dsio/xslx.go

```diff
@@ -0,0 +1,319 @@
+package dsio
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/360EntSecGroup-Skylar/excelize"
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/dataset/vals"
+)
+
+// XLSXReader implements the RowReader interface for the XLSX data format
+type XLSXReader struct {
+	err       error
+	st        *dataset.Structure
+	sheetName string
+	file      *excelize.File
+	r         *excelize.Rows
+	idx       int
+	types     []string
+}
+
+// NewXLSXReader creates a reader from a structure and read source
+func NewXLSXReader(st *dataset.Structure, r io.Reader) (*XLSXReader, error) {
+	// TODO - handle error
+	_, types, _ := terribleHackToGetHeaderRowAndTypes(st)
+
+	rdr := &XLSXReader{
+		st:    st,
+		types: types,
+	}
+
+	// xlsxr := xlsx.NewReader(ReplaceSoloCarriageReturns(r))
+	rdr.file, rdr.err = excelize.OpenReader(r)
+	if rdr.err != nil {
+		return rdr, rdr.err
+	}
+
+	if fcg, err := dataset.ParseFormatConfigMap(dataset.XLSXDataFormat, st.FormatConfig); err == nil {
+		if opts, ok := fcg.(*dataset.XLSXOptions); ok {
+			rdr.sheetName = opts.SheetName
+		}
+	}
+	if rdr.sheetName == "" {
+		rdr.sheetName = "Sheet1"
+	}
+
+	if rdr.err == nil {
+		rdr.r, rdr.err = rdr.file.Rows(rdr.sheetName)
+	}
+
+	return rdr, nil
```

Comment:
I'm unsure, but should this be `return rdr, rdr.err`? Otherwise the error from `Rows` is being ignored.